metadata/src/main/java/org/apache/kafka/controller/FeatureControlManager.java

```diff
@@ -188,12 +182,16 @@ private ApiError updateFeature(
             }
         }
 
-        if (currentVersion != null && newVersion < currentVersion) {
+        if (newVersion < currentVersion) {
             if (upgradeType.equals(FeatureUpdate.UpgradeType.UPGRADE)) {
```

Comment:
Not from this patch, but I was trying to trace back where the `UpgradeType` is determined. I see logic in `QuorumController.updateFeatures` to convert from the code in the request, but it doesn't look like we are checking for the `UNKNOWN` type. Am I missing the validation somewhere?

Reply:
We do check it in `updateFeature`, right?

```
        if (upgradeType.equals(FeatureUpdate.UpgradeType.UNKNOWN)) {        
            return invalidUpdateVersion(featureName, newVersion,            
                "The controller does not support the given upgrade type."); 
        }                                                                   
```

I do kind of wish we did less in `QuorumController#updateFeatures` and more in the `FeatureControlManager` (because it's easier to unit test the latter) but this particular case is covered, I think.

Reply:
Ah, ok. I missed the check.